src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        var data = new UserData(userId, "Details for " + userId);
+        return CompletableFuture.supplyAsync(() -> {
+            cache.put(userId, data);
```

Comment:
The 'cache.put' operation is not thread-safe in the context of a CompletableFuture, as multiple threads may try to update the cache concurrently. To ensure thread safety, consider using 'cache.computeIfAbsent' which is an atomic operation.